core/examples/http_bench.js

```diff
@@ -1,11 +1,77 @@
 // This is not a real HTTP server. We read blindly one time into 'requestBuf',
 // then write this fixed 'responseBuf'. The point of this benchmark is to
 // exercise the event loop in a simple yet semi-realistic way.
-const OP_LISTEN = 1;
-const OP_ACCEPT = 2;
-const OP_READ = 3;
-const OP_WRITE = 4;
-const OP_CLOSE = 5;
+const registry = {};
+
+function initOps(opsMap) {
+  for (const [name, opId] of Object.entries(opsMap)) {
+    const op = registry[name];
+
+    if (!op) {
+      continue;
+    }
+
+    op.setOpId(opId);
+  }
+}
+
+class Op {
+  constructor(name) {
+    if (typeof registry[name] !== "undefined") {
+      throw new Error(`Duplicate op: ${name}`);
+    }
+
+    this.name = name;
+    this.opId = 0;
+    registry[name] = this;
+  }
+
+  setOpId(opId) {
+    this.opId = opId;
+  }
+
+  static handleAsyncMsgFromRust(opId, buf) {
+    const record = recordFromBuf(buf);
+    const { promiseId } = record;
+    const p = promiseMap.get(promiseId);
+    promiseMap.delete(promiseId);
+    p.resolve(record);
+  }
+
+  static sendSync(opId, arg, zeroCopy) {
+    const buf = send(0, opId, arg, zeroCopy);
```

Comment:
I don't like that op_id 0 has two different meanings...

Reply:
How so? Only meaning is that op with id 0 is `get_op_map` op

Reply:
Oh sorry, I thought the first argument of send() here was the op id. Ignore my comment.